wire-ios/Wire-iOS/Sources/AuthenticatedRouter.swift

```diff
@@ -57,14 +56,13 @@ final class AuthenticatedRouter {
     // MARK: - Init
 
     init(
-        rootViewController: UIViewController,
+        rootViewController: @escaping () -> UIViewController,
```

Comment:
I may have missed this point but why do we need to make it a closure?

Reply:
Yep but is the closure required ? I guess we could just do: `init(rootViewcontroller: UIViewController` instead of having `@escaping () -> UIViewController`, from my understanding we're just passing the rootViewController (which is a `UIViewController`) through the init so this would be sufficient, no?

Reply:
You're right I guess, I reverted the changes for a few classes.
Thank you!

Reply:
I am not too comfortable having every router be able to access everything related to the window, what do you think about conforming the window to a `Rootable`protocol that exposes a `rootViewController` that returns the main window root view controller. Router will only be able to access that property and nothing else.

Reply:
> Routers will only be able to access that property and nothing else.

But that was exactly the purpose of the closure I initially had. It was a way to just get the current `rootViewController` value.

Anyway, we have to refactor this sooner or later.
We should have a proper way for presenting alerts etc.
There should be a single point where it is decided how/where alerts and modals are presented.

Reply:
I guess you still don't need the closure, something like this:

``` 
protocol Rootable {
    var rootViewController: UIViewController
}
```

```
extension UIWindow: Rootable {
    var rootViewController: UIViewController { rootViewController }
}

```

```
init(
         rootable: Rootable,
         sessionManager: SessionManager,
         appStateCalculator: AppStateCalculator
     ) {
```

and at callsite you just pass the `mainWindow`

Reply:
But if this will be refactored later, I guess passing the window directly is fine for now